packages/payment/src/models/payment-session.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  BeforeCreate,
+  Entity,
+  Enum,
+  ManyToOne,
+  OneToOne,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import { DAL } from "@medusajs/types"
+import { generateEntityId } from "@medusajs/utils"
+
+import PaymentCollection from "./payment-collection"
+import Payment from "./payment"
+
+/**
+ * @enum
+ *
+ * The status of a payment session.
+ */
+export enum PaymentSessionStatus {
+  /**
+   * The payment is authorized.
+   */
+  AUTHORIZED = "authorized",
+  /**
+   * The payment is pending.
+   */
+  PENDING = "pending",
+  /**
+   * The payment requires an action.
+   */
+  REQUIRES_MORE = "requires_more",
+  /**
+   * An error occurred while processing the payment.
+   */
+  ERROR = "error",
+  /**
+   * The payment is canceled.
+   */
+  CANCELED = "canceled",
+}
+
+type OptionalPaymentSessionProps =
+  | "currency_code"
+  | "data"
+  | "is_selected"
+  | "authorised_at"
+  | DAL.EntityDateColumns
+
+@Entity({ tableName: "payment-session" })
```

Comment:
todo: _snake_casify_
```suggestion
@Entity({ tableName: "payment_session" })
```